config/cmake/H5pubconf.h.in

```diff
@@ -775,25 +748,4 @@
 
 #cmakedefine WORDS_BIGENDIAN @H5_WORDS_BIGENDIANR@
 
-/* Number of bits in a file offset, on hosts where this is settable. */
-#cmakedefine H5__FILE_OFFSET_BITS
-
-/* Define for large files, on AIX-style hosts. */
-#cmakedefine H5__LARGE_FILES
-
-/* Define to empty if `const' does not conform to ANSI C. */
-#cmakedefine H5_const
-
-/* Define to `long int' if <sys/types.h> does not define. */
-#cmakedefine H5_off_t
-
-/* Define to `long' if <sys/types.h> does not define. */
-#cmakedefine H5_ptrdiff_t
-
-/* Define to `unsigned long' if <sys/types.h> does not define. */
-#cmakedefine H5_size_t
-
-/* Define to `long' if <sys/types.h> does not define. */
-#cmakedefine H5_ssize_t
-
 #endif
```

Comment:
A couple of things here:

- This file is a part of the public API, so removing symbols triggers binary compatibility problems and merges to other branches need to be done carefully.
- This file should be kept in sync with the autotools. If it's removed in CMake, it needs to be removed in configure.ac as well.
- HDF5 is supported on systems with dodgy C implementations so we tend to be conservative about ripping out old work-arounds and hacks.

I'm definitely excited to remove old cruft, but we do need to be careful when doing so. At the very least, we need to make sure that CMake and the Autotools stay in sync.

Reply:
I grepped the entire codebase for `H5_ssize_t` and this is the only occurrence anywhere. That's why I removed it, figured it was leftover from a previous purge.

Reply:
Removing cruft from old operating systems and compilers is a never-ending task :)

Reply:
So is this bit ok?

Reply:
I would probably split this into two PRs: One PR with the internal inline code changes. Another with the CMake changes.

The internal inline code PR can be propagated (more) easily to other branches without worrying about binary compatibility issues.

The CMake PR will probably only be able to go to develop (1.13+) as we don't remove public symbols in maintenance branches. We do need to keep the Autotools and CMake in sync, but if you create a PR with the CMake changes, I'll create a PR with the Autotools changes that you can merge into your PR for Larry to do a squash merge.

Reply:
I feel like we should also remove the CMake code that checks for these things, to fully expunge the cruft.

Reply:
Example: The inline checks in cmake_ext_mod/HDFTests.c

Reply:
ok split to: https://github.com/HDFGroup/hdf5/pull/431 let's discuss there